persist/sqlite/consensus_test.go

```diff
@@ -2509,3 +2509,48 @@ func TestMetricCirculatingSupply(t *testing.T) {
 		testutil.Equal(t, "circulating supply", circulatingSupply, metrics.CirculatingSupply)
 	}
 }
+
+func TestDeletedOrphanedTransaction(t *testing.T) {
+	_, _, cm, db := newStore(t, false, nil)
+	genesisState := cm.TipState()
+
+	txn := types.Transaction{
+		ArbitraryData: [][]byte{{0}},
+	}
+	if err := cm.AddBlocks([]types.Block{testutil.MineBlock(cm.TipState(), []types.Transaction{txn}, types.VoidAddress)}); err != nil {
+		t.Fatal(err)
+	}
+	syncDB(t, db, cm)
+
+	{
+		txns, err := db.Transactions([]types.TransactionID{txn.ID()})
+		if err != nil {
+			t.Fatal(err)
+		}
+		testutil.Equal(t, "len(txns)", 1, len(txns))
+		testutil.CheckTransaction(t, txn, txns[0])
+	}
+
+	{
+		// mine to trigger a reorg
```

Comment:
could be a helper

Reply:
Same thing further down in the diff. We could use a helper that starts mining and applying blocks starting from a certain state.
Ideally this whole block would be a single line of code. The previous and next blocks could also be a helper that you declare inline. Then you'd end up with something like

```go
assertTransactions([]types.Transaction{txn})
mineBlocks(genesisState, 2)
assertTransactions([]types.Transacdtion{}) 
```